web/models/notes.ex

```diff
@@ -20,4 +21,15 @@ defmodule Notox.Note do
     |> cast(params, @optional_fields)
     |> validate_required(@required_fields)
   end
+
+  defp title_contains(%Note{title: title}, keyword) do
+    title 
+    |> String.downcase
+    |> String.contains?(String.downcase(keyword))
+  end
+
+  def search(query, keyword) do
+    query
+    |> Enum.filter(fn note -> title_contains(note, keyword) end)
```

Comment:
You want to do as much work as possible in SQL, since it tends to be (although isn't always) much faster than doing the work in whatever programming language you are using. Especially when you have an index on whatever you are searching on. I'd do the filtering with Ecto. I might suggest looking into [`ilike`](https://hexdocs.pm/ecto/Ecto.Query.API.html#ilike/2) ([postgres docs](https://www.postgresql.org/docs/9.0/static/functions-matching.html))